src/mlpack/core/arma_extend/Mat_extra_bones.hpp

```diff
@@ -1,26 +0,0 @@
-// Copyright (C) 2008-2016 National ICT Australia (NICTA)
-//
-// This Source Code Form is subject to the terms of the Mozilla Public
-// License, v. 2.0. If a copy of the MPL was not distributed with this
-// file, You can obtain one at http://mozilla.org/MPL/2.0/.
-// -------------------------------------------------------------------
-//
-// Written by Conrad Sanderson - http://conradsanderson.id.au
-// Written by Ryan Curtin
-
-//! Add a serialization operator.
-template<typename Archive>
-void serialize(Archive& ar, const unsigned int version);
-
-/**
- * These will help us refer the proper vector / column types, only with
- * specifying the matrix type we want to use.
- */
-
-typedef Col<elem_type>   vec_type;
-typedef Col<elem_type>   col_type;
-typedef Row<elem_type>   row_type;
```

Comment:
I think we will still need to keep these, otherwise things won't build.  I'll open a separate PR later to fix that. :+1:

Reply:
@rcurtin I agree, we need to keep them, but since they are no longer part of armadillo internal, are you going to open a pull request on armadillo side? or here? I have just moved them to `serialize_armadillo.hpp`
In fact, my question is how to rename them in the most appropriate way to keep the build, maybe your PR will resolve the issue,